tests/test_nodes.py

```diff
@@ -2,19 +2,57 @@
 from decimal import Decimal, getcontext
 from http import HTTPStatus
 import json
+from typing import Dict, Any, Callable
 import urllib3
 
 import warnings
 
 
-def test_nodes():
+def validate_rpc(rpc: Dict[str, Any], method: str, validator: Callable[[Dict[str, Any]], None]) -> None:
+    """Validate RPC endpoint with specified method and validation function"""
+    try:
+        resp = urllib3.request(
+            'POST',
+            rpc['endpoint'],
+            json={'method': method, 'params': [], 'id': 1, 'jsonrpc': '2.0'},
+            retries=1,
+            timeout=5,
+        )
+    except urllib3.exceptions.MaxRetryError as e:
+        warnings.warn(f'Failed to connect to {rpc["endpoint"]} due to {e}')
+        return
+
+    if resp.status in (HTTPStatus.TOO_MANY_REQUESTS, HTTPStatus.GATEWAY_TIMEOUT):
+        warnings.warn(f'Failed to connect to {rpc["endpoint"]} due to {resp.status}')
+        return
+    
+    try:
+        response_data = resp.json()
+        validator(response_data)
```

Comment:
```suggestion
        assert validator(resp.json()) is True
```
This should also be doing an assertion here right?